src/bonds.jl

```diff
@@ -50,3 +49,40 @@ function _evaluate_env(env::CylindricalBondEnvelope, X::AbstractState)
    return ( (z/zcuteff)^2 - 1 )^(env.pz) * (abs(z) <= zcuteff) * 
             ( (r/env.rcut)^2 - 1 )^(env.pr) * (r <= env.rcut)
 end
+
+
+struct ElipsoidBondEnvelope{T} <: BondEnvelope{T}
+   r0cut::T
+   rcut::T
+   zcut::T  # must satisfy zcut >= r0cut/2
+   p0::Int
+   pr::Int
+   floppy::Bool
+   λ::T
+end
+
+ElipsoidBondEnvelope(r0cut, zcut, rcut; p0=2, pr=2, floppy=false, λ=.5) = ElipsoidBondEnvelope(r0cut, zcut, rcut, p0, pr, floppy, λ)
+ElipsoidBondEnvelope(r0cut, cut; p0=2, pr=2, floppy=false, λ=.5) = ElipsoidBondEnvelope(r0cut, cut, cut, p0, pr, floppy, λ)
+
+function _evaluate_bond(env::ElipsoidBondEnvelope, X::AbstractState)
+   r = norm(X.rr0)
+   return ((r/env.r0cut)^2 - 1)^(env.p0) * (r <= env.r0cut)
+end
+
+function _evaluate_env(env::ElipsoidBondEnvelope, X::AbstractState)
+   # convert to cylindrical coordinates
+   x_centre = env.λ * X.rr0
+   x̃ = X.rr - x_centre
+   if norm(X.rr0) > 0
+      x_centre_0  = X.rr0/norm(X.rr0)
+      z̃ = dot(x_centre_0, x̃ )
+      r̃ = norm( x̃ - z̃ * x_centre_0)
+   else
+      z̃ = 0.0
+      r̃ = norm(x̃)
+   end
+   # then return the correct cutoff
```

Comment:
I can try that of course. May I also work in this PR or start a new one? (Or maybe Matthias has already started implementing that? I notice this comment just because I am now using the code in this PR for implementing bond basis for offsite Hamiltonians...)